<launch>
    <!-- <arg name="robot_name"/> -->
    <arg name="init_pose" doc="will be in the shape -x X -y Y -z Z"/>
    <arg name="role" doc="[TOI, AT, DT]"/>
    <arg name="robot_number" default="0"/>

    <arg if="$(eval role == 'TOI')" 
    name="robot_name" value="TOI" />
    <arg unless="$(eval role == 'TOI')" 
    name="robot_name" value="$(arg role)$(arg robot_number)" />

    <arg if="$(eval role == 'TOI')" 
    name="robot_description" value="/TOI/robot_description" />
    <arg if="$(eval role == 'TOI')" 
    name="orientation" value="-R 0.0 -P 0.0 -Y 0.0" />
    <arg if="$(eval role == 'DT')" 
    name="robot_description" value="/DT$(arg robot_number)/robot_description" />
    <arg if="$(eval role == 'DT')" 
    name="orientation" value="-R 0.0 -P 0.0 -Y 0.0" />
    <arg if="$(eval role == 'AT')" 
    name="robot_description" value="/AT$(arg robot_number)/robot_description" />
    <arg if="$(eval role == 'AT')" 
    name="orientation" value="-R 0.0 -P 0.0 -Y 3.1415" />

    <node   name="spawn_robot" 
            pkg="gazebo_ros" 
            type="spawn_model"
            args="  $(arg init_pose)
                    $(arg orientation)
                    -urdf 
                    -param $(arg robot_description)
                    -model $(arg robot_name)"
            respawn="false" 
            output="screen" />
    <!--<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />-->
</launch>